File: Bit_Manipulation/191_number_of_1_bits/191.cpp
#include <stdint.h>
using namespace std;

class Solution {
public:
    int hammingWeight(uint32_t n) {
        int sum = 0;
        while (n != 0) {
            sum += 1;
            n &= (n - 1);
        }
        return sum;
    }
};

File: Bit_Manipulation/191_number_of_1_bits/191.rs


impl Solution {
    pub fn hammingWeight (n: u32) -> i32 {
        let mut sum: i32 = 0;
        let mut n = n;
        while n != 0 {
            sum += 1;
            n &= (n - 1);
        }
        return sum;
    }
}